### PR TITLE
Helper method to allow easier creation of AudioSources

### PR DESCRIPTION
**Pull request by Peepo-Juice**
With the current state of things, this is a simple way to spawn an AudioSource component:
```rust
fn startup(mut commands: Commands, studio: Res<FmodStudio>) {
    let event_description = studio.0.get_event("event:/Music/Level 03").unwrap();

    commands
        .spawn(MyMusicPlayer)
        .insert(AudioSource::new(event_description));
}
```

With my proposed helper method, it would look more like this:
```rust
fn startup(mut commands: Commands, studio: Res<FmodStudio>) {
    commands
        .spawn(MyMusicPlayer)
        .insert(studio.build_audio_source_from_path("event:/Music/Level 03"));
}
```

I have some questions, but I will leave them as inline comments for further discussion (I never tried doing that before so we shall see how it goes).

### CODE REVIEW

File: src/fmod_studio.rs [hunk 1]
```diff
@@ -2,6 +2,7 @@ use std::env::var;
 use std::fs::canonicalize;
 use std::path::{Path, PathBuf};
 
+use crate::components::audio_source::AudioSource as BevyFmodAudioSource;
```
Comment (Peepo-Juice) [inline src/fmod_studio.rs:5]: Had a naming conflict with AudioSource. Feel free to handle this differently or call change the alias.

File: src/fmod_studio.rs [hunk 2]
```diff
@@ -66,4 +67,12 @@ impl FmodStudio {
 
         studio
     }
+
+    pub fn build_audio_source_from_path(&self, path: &str) -> BevyFmodAudioSource {
```
Comment (Peepo-Juice) [inline src/fmod_studio.rs:71]: This name could be better. I can't think of anything better to save my life.

File: src/fmod_studio.rs (hunk 1 above)
Comment (GitGhillie) [reply src/fmod_studio.rs:5]: I don't seem to have this problem with just `use crate::components::audio_source::AudioSource;`, please try again :P

File: src/fmod_studio.rs (hunk 2 above)
Comment (GitGhillie) [reply src/fmod_studio.rs:71]: Since `path` is already in the arguments I think it's justified to call it `build_audio_source`

File: .gitignore [hunk 3]
```diff
@@ -6,3 +6,4 @@ assets/
 *.so.*
 demo_project/
 Cargo.lock
+.vscode/
```
Comment (GitGhillie) [inline .gitignore:9]: Personally I agree with this, but iirc @Salzian had some reservations about this

Comment (GitGhillie) [reply .gitignore:9]: Maybe add both like in the Bevy repo:
```
/.idea
/.vscode
```

File: src/fmod_studio.rs [hunk 4]
```diff
@@ -66,4 +67,12 @@ impl FmodStudio {
...
+        let event_description = self
+            .0
+            .get_event(path)
+            .expect("The event is expected to exist in the FMOD project.");
```
Comment (GitGhillie) [inline src/fmod_studio.rs:75]:
```suggestion
            .expect("Failed to get event from path or ID");
```

**Review by GitGhillie — CHANGES_REQUESTED** [review]
I'm not a 100% sure if we are going in the right direction with this helper method but I would like to see it merged until we find a better solution.

Please update the examples to use this helper method as well, and if you update the spatial example you will see there is a bit more work to be done :)

File: src/fmod_studio.rs (hunk 1 above)
Comment (Peepo-Juice) [reply src/fmod_studio.rs:5]: You are correct, I am a fool.

File: .gitignore (hunk 3 above)
Comment (Peepo-Juice) [reply .gitignore:9]: I don't see a problem with it, but id love to see the conversation with @Salzian if its public.

Comment (Salzian) [reply .gitignore:9]: `.vscode` and `.idea` sometimes serve valuable metadata about the repository between developers. They usually have their own nested `.gitignore` to only share the important bits. So feel free to check in the `.vscode` folder.

Comment (Peepo-Juice) [reply .gitignore:9]: I'm gonna undo the changes, feels out of scope for the PR.

**Comment by Salzian** [comment]
Closing for now until we have a better asset management system.

**Pull request closed without merging**